test/e2e_node/eviction_test.go

```diff
@@ -0,0 +1,567 @@
+		}
+		Eventually(func() error {
+			logFunc()
+			if pressure == nil || hasPressure(f, pressure) {
+				return nil
+			}
+			return fmt.Errorf("Condition: %s not encountered", testCondition)
+		}, pressureTimeout, evictionPollInterval).Should(BeNil())
+
+		Eventually(func() error {
+			// Gather current information
+			updatedPodList, err := f.ClientSet.Core().Pods(f.Namespace.Name).List(metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			updatedPods := updatedPodList.Items
+			for _, p := range updatedPods {
+				framework.Logf("fetching pod %s; phase= %v", p.Name, p.Status.Phase)
+			}
+			logKubeletMetrics(kubeletmetrics.EvictionStatsAgeKey)
+			logFunc()
+			if pressure != nil {
+				hasPressure(f, pressure)
+			}
+
+			By("checking eviction ordering and ensuring important pods dont fail")
+			done := true
+			for _, priorityPodSpec := range podTestSpecs {
+				var priorityPod v1.Pod
+				for _, p := range updatedPods {
+					if p.Name == priorityPodSpec.pod.Name {
+						priorityPod = p
+					}
+				}
+				Expect(priorityPod).NotTo(BeNil())
+
+				// Check eviction ordering.
+				// Note: it is alright for a priority 1 and priority 2 pod (for example) to fail in the same round
+				for _, lowPriorityPodSpec := range podTestSpecs {
+					var lowPriorityPod v1.Pod
+					for _, p := range updatedPods {
+						if p.Name == lowPriorityPodSpec.pod.Name {
+							lowPriorityPod = p
+						}
+					}
+					Expect(lowPriorityPod).NotTo(BeNil())
+					if priorityPodSpec.evictionPriority < lowPriorityPodSpec.evictionPriority && lowPriorityPod.Status.Phase == v1.PodRunning {
+						Expect(priorityPod.Status.Phase).NotTo(Equal(v1.PodFailed),
+							fmt.Sprintf("%s pod failed before %s pod", priorityPodSpec.pod.Name, lowPriorityPodSpec.pod.Name))
+					}
+				}
+
+				// EvictionPriority 0 pods should not fail
+				if priorityPodSpec.evictionPriority == 0 {
+					Expect(priorityPod.Status.Phase).NotTo(Equal(v1.PodFailed),
+						fmt.Sprintf("%s pod failed (and shouldn't have failed)", priorityPod.Name))
+				}
+
+				// If a pod that is not evictionPriority 0 has not been evicted, we are not done
+				if priorityPodSpec.evictionPriority != 0 && priorityPod.Status.Phase != v1.PodFailed {
+					done = false
+				}
+			}
+			if done {
+				return nil
+			}
+			return fmt.Errorf("pods that caused %s have not been evicted.", testCondition)
+		}, pressureTimeout, evictionPollInterval).Should(BeNil())
+
+		// We observe pressure from the API server.  The eviction manager observes pressure from the kubelet internal stats.
+		// This means the eviction manager will observe pressure before we will, creating a delay between when the eviction manager
+		// evicts a pod, and when we observe the pressure by querrying the API server.  Add a delay here to account for this delay
+		By("making sure pressure from test has surfaced before continuing")
+		time.Sleep(pressureDelay)
```

Comment:
I havent found a way to do this effectively.  The pressure may have already subsided after the last pod is evicted, so we cant wait to eventually see it.

Reply:
Why not poll for pressure with a timeout of `pressureDelay`, so that at least your best-case (you see pressure and skip the rest of the sleep) improves. Maybe this is getting to into the optimization weeds.

Reply:
Yeah, I am not sure saving ~10 seconds on a 10 minute test is worth it here.  It doesnt affect correctness, so I am inclined to leave as-is.

Reply:
sgtm